pyteal/ast/binaryexpr.py

```diff
@@ -259,3 +259,185 @@ def GetByte(value: Expr, index: Expr) -> BinaryExpr:
         index: The index of the byte to extract. Must evaluate to an integer less than Len(value).
     """
     return BinaryExpr(Op.getbyte, (TealType.bytes, TealType.uint64), TealType.uint64, value, index)
+
+def BAdd(left: Expr, right: Expr) -> BinaryExpr:
+    """Add two numbers as bytes.
+    
+    Produces left + right.
```

Comment:
Can you elaborate on how the byte slices are interpreted as integers for each of these descriptions, like "Produces left + right, where left and right are interpreted as big-endian unsigned integers." It would be good to mention that the arguments must be 64 bytes or less too.

Also, for &, |, ^, it's worth mentioning that the arguments are "zero-left extended to the greater of their lengths" (taken from https://github.com/algorandfoundation/specs/blob/master/dev/TEAL_opcodes.md#b-11).